### PR TITLE
[MacCatalyst] Generalize a test for Apple Silicon and new iOS version…

### PR DESCRIPTION
**Pull request by dcci**
…(s).

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci test

**Review by adrian-prantl — APPROVED** [review]
Thanks!

File: lldb/test/API/lang/swift/macCatalyst/TestSwiftMacCatalyst.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def test_macCatalyst(self):
         lldbutil.run_to_source_breakpoint(self, "break here",
                                           lldb.SBFileSpec('main.swift'))
         self.expect("image list -t -b",
-                    patterns=["x86_64-apple-ios13.0.0-macabi a\.out",
+                    patterns=[arch + ".*-apple-ios.*-macabi a\.out",
                               "x86_64.*-apple-ios.*-macabi Foundation",
```
Comment (adrian-prantl) [inline lldb/test/API/lang/swift/macCatalyst/TestSwiftMacCatalyst.py:33]: Why don't we need to change this line? It also has x86 hardcoded.

Comment (dcci) [reply lldb/test/API/lang/swift/macCatalyst/TestSwiftMacCatalyst.py:33]: Good question ;) I just cherry picked your commit from our internal branch. I'll change that too

**Comment by dcci** [comment]
@swift-ci test

**Comment by dcci** [comment]
@swift-ci test

**Comment by dcci** [comment]
Merging

**Pull request merged**